client.go

```diff
@@ -138,18 +138,24 @@ func (client *Client) Close() error {
 	defer client.lock.Unlock()
 	Logger.Println("Closing Client")
 
+	var wg sync.WaitGroup
 	for _, broker := range client.brokers {
-		safeAsyncClose(broker)
+		asyncCloseWithAck(broker, &wg)
 	}
+
 	client.brokers = nil
 	client.metadata = nil
 
 	if client.seedBroker != nil {
-		safeAsyncClose(client.seedBroker)
+		asyncCloseWithAck(client.seedBroker, &wg)
 	}
 
 	close(client.closer)
 
+	/* Unsure all brokers have closed before exitting so the caller
```

Comment:
This comment is a little hard to understand. Is "Unsure" supposed to be "Wait for"?

s/exitting/exiting/